### PR TITLE
Refactor: chat UI

### PR DESCRIPTION
**Pull request by marwan2232004**

### CODE REVIEW

File: lib/features/chat/view/screens/pinned_messages_screen.dart [hunk 1]
```diff
@@ -53,9 +53,9 @@ class _PinnedMessagesScreen extends ConsumerState<PinnedMessagesScreen>
   int _currentMatch = 1;
 
   // ignore: unused_field
-  int _currentMatchIndex = 0;
-  Map<int, List<MapEntry<int, int>>> _messageMatches = {};
-  List<int> _messageIndices = [];
+  final int _currentMatchIndex = 0;
+  final Map<int, List<MapEntry<int, int>>> _messageMatches = {};
+  final List<int> _messageIndices = [];
```
Comment (Ahmed-Aladdiin) [inline lib/features/chat/view/screens/pinned_messages_screen.dart:58]: I don't think that these should be 'final'
please, check them with @Bishoywadea

File: lib/features/chat/view/widget/audio_message_widget.dart [hunk 2]
```diff
@@ -136,18 +134,98 @@ class AudioMessageWidgetState extends State<AudioMessageWidget>
...
+      topLeft: Radius.circular(topLeft ? widget.borderRadius : 0),
+      bottomLeft: Radius.circular(bottomLeft ? widget.borderRadius : 0),
+      topRight: Radius.circular(topRight ? widget.borderRadius : 0),
+      bottomRight: Radius.circular(bottomRight ? widget.borderRadius : 0),
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
+    Widget audioWaveform = AudioFileWaveforms(
```
Comment (Ahmed-Aladdiin) [inline lib/features/chat/view/widget/audio_message_widget.dart]: you don't put them this way in the build method, instead do it this way:
```dart
Widget build(BuildContext context) {
return Container(child: _subwidget(context))
}
Widget _subWidget(BuildContext context) {
return SizedBox();
}
```

File: lib/features/chat/view/screens/pinned_messages_screen.dart (hunk 1 above)
Comment (marwan2232004) [reply lib/features/chat/view/screens/pinned_messages_screen.dart:58]: In the current logic, these variables are not being set in the code but in the future, they should be changed.

**Review by Ahmed-Aladdiin — CHANGES_REQUESTED** [review]
Just remove those long ui variables from the build method, and ask Bishoy about whether these variable should be final or not. 

probably not

**Pull request merged**